src/SonarScanner.MSBuild.PreProcessor/WebClientDownloaderBuilder.cs

```diff
@@ -95,6 +113,24 @@ public WebClientDownloader Build()
         return new WebClientDownloader(client, baseAddress, logger);
     }
 
+    private static bool ServerCertificateCustomValidationCallback(X509Certificate2Collection trustStore,
+        HttpRequestMessage message, X509Certificate2 certificate, X509Chain chain, SslPolicyErrors errors)
+    {
+        if (errors is SslPolicyErrors.None)
+        {
+            return true;
+        }
+        if (errors is SslPolicyErrors.RemoteCertificateChainErrors // Don't do HasFlags. Any other errors than RemoteCertificateChainErrors should fail the handshake.
+            && chain.ChainStatus.All(x => x.Status is X509ChainStatusFlags.UntrustedRoot)) // Any other violations than UntrustedRoot should fail the handshake.
+        {
+            if (trustStore.Find(X509FindType.FindBySerialNumber, certificate.SerialNumber, validOnly: false) is { Count: > 0 } certificatesInTrustStore)
```

Comment:
Why did you accept this [issue](https://sonarcloud.io/project/issues?issues=AZS3U8lvBccbvNi9y63X&open=AZS3U8lvBccbvNi9y63X&pullRequest=2306&id=sonarscanner-msbuild)?

The ifs could be merged from my understanding.
```suggestion
        if (errors is SslPolicyErrors.RemoteCertificateChainErrors // Don't do HasFlags. Any other errors than RemoteCertificateChainErrors should fail the handshake.
            && chain.ChainStatus.All(x => x.Status is X509ChainStatusFlags.UntrustedRoot) // Any other violations than UntrustedRoot should fail the handshake.
            && trustStore.Find(X509FindType.FindBySerialNumber, certificate.SerialNumber, validOnly: false) is { Count: > 0 } certificatesInTrustStore)
        {
```

Reply:
![image](https://github.com/user-attachments/assets/d6f9d944-c0bf-426b-a2dd-efe1da4c8039)